pkg/image/extensions.go

```diff
@@ -0,0 +1,94 @@
+package image
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+type Extensions struct {
+	Config         ExportConfig
+	GithubEndpoint GithubEndpoint
+}
+
+type Release struct {
+	TagName string `json:"tag_name"`
+}
+
+type GithubEndpoint struct {
+	URL string
+}
+
+var ExtensionEndpoints = []GithubEndpoint{
+	{URL: "https://api.github.com/repos/rancher/ui-plugin-charts/releases"},
+}
+
+func ExtensionConfig(endpointIndex int) Extensions {
+	if endpointIndex >= len(ExtensionEndpoints) {
+		endpointIndex = 0
+	}
+
+	return Extensions{
+		Config:         ExportConfig{},
+		GithubEndpoint: ExtensionEndpoints[endpointIndex],
+	}
+}
```

Comment:
I feel this extra layer of abstraction is unnecessary, and it is confusing to use `ExtensionConfig(0)` to get the hardcoded GithubEndpoint for ui-plugin-charts; what if someone adds a new entry to the beginning of the list ExtensionEndpoints? The code still works but returns a wrong result. 

My suggestion is: 
1. Drop the type `ExtensionEndpoints` and the function `ExtensionConfig` 
2. Make a new `Extensions` instance in reslove.go directly, something like this 
```
	// fetch images from extension catalog images
	extensions := ExtensionConfig{... ... }
	if err := extensions.FetchExtensionImages(imagesSet); err != nil {
		return nil, nil, errors.Wrap(err, "failed to fetch images from extensions")
	}
```

Reply:
Good points. I removed the type and function, and fixed the `FetchExtensionImages` to iterate correctly through the `ExtensionEndpoints` array.

Let me know what you think.